### ISSUE TITLE
Fails to build 2.4.1 on Debian Buster (stable)

### ISSUE THREAD

**Issue by trebmuh** [audacity/audacity#531]
I took the tarball [from fosshub](https://www.fosshub.com/Audacity.html).
Trying to build from it with a cmake 3.16.3 (following #529), it fails with:

```
[ 57%] Building CXX object src/CMakeFiles/Audacity.dir/AudioIO.cpp.o
cd /build/audacity-2.4.1/obj-x86_64-linux-gnu/src && /usr/bin/c++  -DAudacity_EXPORTS -DBUILDING_AUDACITY -DCMAKE -DHAVE_LRINT -DHAVE_LRINTF -DHAVE_MLOCK -DHAVE_SYS_SOUNDCARD_H=1 -DPA_USE_ALSA=1 -DPA_USE_JACK=1 -DPA_USE_OSS=1 -DWXINTL_NO_GETTEXT_MACRO -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/src/private -I/build/audacity-2.4.1/lib-src/lib-widget-extra -I/build/audacity-2.4.1/include -I/build/audacity-2.4.1/src -I/usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -I/build/audacity-2.4.1/lib-src/FileDialog -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lame/public -I/build/audacity-2.4.1/lib-src/portaudio-v19/include -I/build/audacity-2.4.1/lib-src/libid3tag -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/libmad/public -I/build/audacity-2.4.1/lib-src/libmad -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lv2/public -I/build/audacity-2.4.1/lib-src/lv2/lv2 -I/build/audacity-2.4.1/lib-src/lv2/lilv -I/build/audacity-2.4.1/lib-src/lv2/serd -I/build/audacity-2.4.1/lib-src/lv2/sord -I/build/audacity-2.4.1/lib-src/lv2/sratom -I/build/audacity-2.4.1/lib-src/lv2/suil -I/build/audacity-2.4.1/lib-src/portsmf/include -I/build/audacity-2.4.1/lib-src/libnyquist -I/build/audacity-2.4.1/lib-src/portmixer/include -I/build/audacity-2.4.1/lib-src/sbsms/include -isystem /usr/include/gtk-2.0 -isystem /usr/lib/x86_64-linux-gnu/gtk-2.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/cairo -isystem /usr/include/pango-1.0 -isystem /usr/include/atk-1.0 -isystem /usr/include/pixman-1 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/harfbuzz -isystem /usr/include/fribidi -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/uuid -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/soundtouch  -g -O2 -fdebug-prefix-map=/build/audacity-2.4.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -pthread -std=gnu++14 -Winvalid-pch -include /build/audacity-2.4.1/obj-x86_64-linux-gnu/src/CMakeFiles/Audacity.dir/cmake_pch.hxx -o CMakeFiles/Audacity.dir/AudioIO.cpp.o -c /build/audacity-2.4.1/src/AudioIO.cpp
/build/audacity-2.4.1/src/AudioIO.cpp: In function 'PaTime PaUtil_GetTime()':
/build/audacity-2.4.1/src/AudioIO.cpp:500:5: error: 'gettimeofday' was not declared in this scope
     gettimeofday( &tv, NULL );
     ^~~~~~~~~~~~
/build/audacity-2.4.1/src/AudioIO.cpp:500:5: note: suggested alternative: 'wxGetTimeOfDay'
     gettimeofday( &tv, NULL );
     ^~~~~~~~~~~~
     wxGetTimeOfDay
[ 57%] Building CXX object src/CMakeFiles/Audacity.dir/AudioIOBase.cpp.o
cd /build/audacity-2.4.1/obj-x86_64-linux-gnu/src && /usr/bin/c++  -DAudacity_EXPORTS -DBUILDING_AUDACITY -DCMAKE -DHAVE_LRINT -DHAVE_LRINTF -DHAVE_MLOCK -DHAVE_SYS_SOUNDCARD_H=1 -DPA_USE_ALSA=1 -DPA_USE_JACK=1 -DPA_USE_OSS=1 -DWXINTL_NO_GETTEXT_MACRO -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/src/private -I/build/audacity-2.4.1/lib-src/lib-widget-extra -I/build/audacity-2.4.1/include -I/build/audacity-2.4.1/src -I/usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -I/build/audacity-2.4.1/lib-src/FileDialog -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lame/public -I/build/audacity-2.4.1/lib-src/portaudio-v19/include -I/build/audacity-2.4.1/lib-src/libid3tag -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/libmad/public -I/build/audacity-2.4.1/lib-src/libmad -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lv2/public -I/build/audacity-2.4.1/lib-src/lv2/lv2 -I/build/audacity-2.4.1/lib-src/lv2/lilv -I/build/audacity-2.4.1/lib-src/lv2/serd -I/build/audacity-2.4.1/lib-src/lv2/sord -I/build/audacity-2.4.1/lib-src/lv2/sratom -I/build/audacity-2.4.1/lib-src/lv2/suil -I/build/audacity-2.4.1/lib-src/portsmf/include -I/build/audacity-2.4.1/lib-src/libnyquist -I/build/audacity-2.4.1/lib-src/portmixer/include -I/build/audacity-2.4.1/lib-src/sbsms/include -isystem /usr/include/gtk-2.0 -isystem /usr/lib/x86_64-linux-gnu/gtk-2.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/cairo -isystem /usr/include/pango-1.0 -isystem /usr/include/atk-1.0 -isystem /usr/include/pixman-1 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/harfbuzz -isystem /usr/include/fribidi -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/uuid -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/soundtouch  -g -O2 -fdebug-prefix-map=/build/audacity-2.4.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -pthread -std=gnu++14 -Winvalid-pch -include /build/audacity-2.4.1/obj-x86_64-linux-gnu/src/CMakeFiles/Audacity.dir/cmake_pch.hxx -o CMakeFiles/Audacity.dir/AudioIOBase.cpp.o -c /build/audacity-2.4.1/src/AudioIOBase.cpp
[ 57%] Building CXX object src/CMakeFiles/Audacity.dir/AutoRecovery.cpp.o
cd /build/audacity-2.4.1/obj-x86_64-linux-gnu/src && /usr/bin/c++  -DAudacity_EXPORTS -DBUILDING_AUDACITY -DCMAKE -DHAVE_LRINT -DHAVE_LRINTF -DHAVE_MLOCK -DHAVE_SYS_SOUNDCARD_H=1 -DPA_USE_ALSA=1 -DPA_USE_JACK=1 -DPA_USE_OSS=1 -DWXINTL_NO_GETTEXT_MACRO -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/src/private -I/build/audacity-2.4.1/lib-src/lib-widget-extra -I/build/audacity-2.4.1/include -I/build/audacity-2.4.1/src -I/usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -I/build/audacity-2.4.1/lib-src/FileDialog -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lame/public -I/build/audacity-2.4.1/lib-src/portaudio-v19/include -I/build/audacity-2.4.1/lib-src/libid3tag -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/libmad/public -I/build/audacity-2.4.1/lib-src/libmad -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lv2/public -I/build/audacity-2.4.1/lib-src/lv2/lv2 -I/build/audacity-2.4.1/lib-src/lv2/lilv -I/build/audacity-2.4.1/lib-src/lv2/serd -I/build/audacity-2.4.1/lib-src/lv2/sord -I/build/audacity-2.4.1/lib-src/lv2/sratom -I/build/audacity-2.4.1/lib-src/lv2/suil -I/build/audacity-2.4.1/lib-src/portsmf/include -I/build/audacity-2.4.1/lib-src/libnyquist -I/build/audacity-2.4.1/lib-src/portmixer/include -I/build/audacity-2.4.1/lib-src/sbsms/include -isystem /usr/include/gtk-2.0 -isystem /usr/lib/x86_64-linux-gnu/gtk-2.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/cairo -isystem /usr/include/pango-1.0 -isystem /usr/include/atk-1.0 -isystem /usr/include/pixman-1 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/harfbuzz -isystem /usr/include/fribidi -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/uuid -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/soundtouch  -g -O2 -fdebug-prefix-map=/build/audacity-2.4.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -pthread -std=gnu++14 -Winvalid-pch -include /build/audacity-2.4.1/obj-x86_64-linux-gnu/src/CMakeFiles/Audacity.dir/cmake_pch.hxx -o CMakeFiles/Audacity.dir/AutoRecovery.cpp.o -c /build/audacity-2.4.1/src/AutoRecovery.cpp
[ 57%] Building CXX object src/CMakeFiles/Audacity.dir/AutoRecoveryDialog.cpp.o
cd /build/audacity-2.4.1/obj-x86_64-linux-gnu/src && /usr/bin/c++  -DAudacity_EXPORTS -DBUILDING_AUDACITY -DCMAKE -DHAVE_LRINT -DHAVE_LRINTF -DHAVE_MLOCK -DHAVE_SYS_SOUNDCARD_H=1 -DPA_USE_ALSA=1 -DPA_USE_JACK=1 -DPA_USE_OSS=1 -DWXINTL_NO_GETTEXT_MACRO -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/src/private -I/build/audacity-2.4.1/lib-src/lib-widget-extra -I/build/audacity-2.4.1/include -I/build/audacity-2.4.1/src -I/usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -I/build/audacity-2.4.1/lib-src/FileDialog -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lame/public -I/build/audacity-2.4.1/lib-src/portaudio-v19/include -I/build/audacity-2.4.1/lib-src/libid3tag -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/libmad/public -I/build/audacity-2.4.1/lib-src/libmad -I/build/audacity-2.4.1/obj-x86_64-linux-gnu/cmake-proxies/lv2/public -I/build/audacity-2.4.1/lib-src/lv2/lv2 -I/build/audacity-2.4.1/lib-src/lv2/lilv -I/build/audacity-2.4.1/lib-src/lv2/serd -I/build/audacity-2.4.1/lib-src/lv2/sord -I/build/audacity-2.4.1/lib-src/lv2/sratom -I/build/audacity-2.4.1/lib-src/lv2/suil -I/build/audacity-2.4.1/lib-src/portsmf/include -I/build/audacity-2.4.1/lib-src/libnyquist -I/build/audacity-2.4.1/lib-src/portmixer/include -I/build/audacity-2.4.1/lib-src/sbsms/include -isystem /usr/include/gtk-2.0 -isystem /usr/lib/x86_64-linux-gnu/gtk-2.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/cairo -isystem /usr/include/pango-1.0 -isystem /usr/include/atk-1.0 -isystem /usr/include/pixman-1 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/harfbuzz -isystem /usr/include/fribidi -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/uuid -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/soundtouch  -g -O2 -fdebug-prefix-map=/build/audacity-2.4.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -pthread -std=gnu++14 -Winvalid-pch -include /build/audacity-2.4.1/obj-x86_64-linux-gnu/src/CMakeFiles/Audacity.dir/cmake_pch.hxx -o CMakeFiles/Audacity.dir/AutoRecoveryDialog.cpp.o -c /build/audacity-2.4.1/src/AutoRecoveryDialog.cpp
make[3]: *** [src/CMakeFiles/Audacity.dir/build.make:187: src/CMakeFiles/Audacity.dir/AudioIO.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

Does that ring a bell for you?

**Comment by lllucius** [comment]
I believe you'll need to rework the "0005-Fix-building-against-the-system-portaudio-library.patch" Debian patch.  Maybe add an include of "<sys/time.h>".

I'll actually be committing a change shortly that should allow you guys to get rid of this patch entirely, but that doesn't help you for the 2.4.1 release.

**Comment by trebmuh** [comment]
This one looks to be closed with 2.4.2 .
Thank you!
